lib/functional-constraints/mutuallyExclusiveFields.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const _ = require('lodash');
+const jsonschema = require('jsonschema');
+
+// NOTE: While it would be possible to accomplish this with a solution like
+//   https://stackoverflow.com/questions/28162509/mutually-exclusive-property-groups#28172831
+//   it was harder to read and understand.
+
+const incompatibleFields = [
```

Comment:
I think this is simpler indeed than the stackoverflow :). 

Would it be helpful if we added comments on why the combinations are not available (comments would suffice)?

Reply:
That's a good point, but screen estate is limited for the errors, there. Do you have any suggestions?

Reply:
Oh I meant inside the code.